test/e2e/apimachinery/apply.go

```diff
@@ -0,0 +1,303 @@
+					framework.Failf("Failed to re-apply object using Apply patch: %v", err)
+				}
+			}
+		}
+	})
+
+	ginkgo.It("PUT requests which change nothing should not change the resource version", func() {
+		podName := "no-op"
+		podResource := "pods"
+		podBytes := []byte(`{
+		"apiVersion": "v1",
+		"kind": "Pod",
+		"metadata": {
+			"name": "` + podName + `",
+			"labels": {
+				"a": "one",
+				"c": "two",
+				"b": "three"
+			}
+		},
+		"spec": {
+			"containers": [{
+				"name":  "test-container-a",
+				"image": "test-image-one"
+			},{
+				"name":  "test-container-c",
+				"image": "test-image-two"
+			},{
+				"name":  "test-container-b",
+				"image": "test-image-three"
+			}]
+		}
+	}`)
+
+		_, err := client.CoreV1().RESTClient().Patch(types.ApplyPatchType).
+			Namespace(ns).
+			Param("fieldManager", "apply_test").
+			Resource(podResource).
+			Name(podName).
+			Body(podBytes).
+			Do(context.TODO()).
+			Get()
+		if err != nil {
+			framework.Failf("Failed to create object: %v", err)
+		}
+
+		// Sleep for one second to make sure that the times of each update operation is different.
+		time.Sleep(1 * time.Second)
+
+		createdObject, err := client.CoreV1().RESTClient().Get().Namespace(ns).Resource(podResource).Name(podName).Do(context.TODO()).Get()
+		if err != nil {
+			framework.Failf("Failed to retrieve created object: %v", err)
+		}
+
+		createdAccessor, err := meta.Accessor(createdObject)
+		if err != nil {
+			framework.Failf("Failed to get meta accessor for created object: %v", err)
+		}
+
+		createdBytes, err := json.MarshalIndent(createdObject, "\t", "\t")
+		if err != nil {
+			framework.Failf("Failed to marshal created object: %v", err)
+		}
+
+		// Test that we can put the same object and don't change the RV
+		_, err = client.CoreV1().RESTClient().Put().
+			Namespace(ns).
+			Resource(podResource).
+			Name(podName).
+			Body(createdBytes).
+			Do(context.TODO()).
+			Get()
+		if err != nil {
+			framework.Failf("Failed to apply no-op update: %v", err)
+		}
+
+		updatedObject, err := client.CoreV1().RESTClient().Get().Namespace(ns).Resource(podResource).Name(podName).Do(context.TODO()).Get()
+		if err != nil {
+			framework.Failf("Failed to retrieve updated object: %v", err)
+		}
+
+		updatedAccessor, err := meta.Accessor(updatedObject)
+		if err != nil {
+			framework.Failf("Failed to get meta accessor for updated object: %v", err)
+		}
+
+		updatedBytes, err := json.MarshalIndent(updatedObject, "\t", "\t")
+		if err != nil {
+			framework.Failf("Failed to marshal updated object: %v", err)
+		}
+
+		if createdAccessor.GetResourceVersion() != updatedAccessor.GetResourceVersion() {
+			framework.Failf("Expected same resource version to be %v but got: %v\nold object:\n%v\nnew object:\n%v",
+				createdAccessor.GetResourceVersion(),
+				updatedAccessor.GetResourceVersion(),
+				string(createdBytes),
+				string(updatedBytes),
+			)
+		}
+	})
+
+	ginkgo.It("Force apply should ignore API errors ", func() {
```

Comment:
this is specifically ignoring conflict errors, right? should probably use that as the test title, not API errors generally